src/v/datalake/table_id_provider.cc

```diff
@@ -39,7 +39,10 @@ table_id_provider::dlq_table_id(const model::topic& t) {
     return {
       // TODO: namespace as a topic property? Keep it in the table metadata?
       .ns = {"redpanda"},
-      .table = fmt::format("{}~dlq", sanitize_topic_name(t)),
+      .table = fmt::format(
+        "{}{}",
+        sanitize_topic_name(t),
+        config::shard_local_cfg().iceberg_dlq_table_suffix()),
```

Comment:
should we sanitize the result (i.e. remove `'.'`) the way we do for the topic name?

Reply:
I'd leave that for the user. Especially since the property has the word "topic" in it. If they tune one because they know of an issue I'd expect them to be careful when choosing a value for the other one too.